src/Rules/Variables/CoalesceRule.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Variables;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\NullType;
+use PHPStan\Type\VerbosityLevel;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\BinaryOp\Coalesce>
+ */
+class CoalesceRule implements \PHPStan\Rules\Rule
+{
+
+	public function getNodeType(): string
+	{
+		return Node\Expr\BinaryOp\Coalesce::class;
```

Comment:
This rule should probably also handle the `AssignOp` version of this.

Reply:
I've add this now, thanks for the feedback!